e2e_test/batch/types/array_ty.slt.part

```diff
@@ -105,3 +105,106 @@ drop table t;
 # This is different from PostgreSQL, we may want to change this in the future.
 statement ok
 select array[array[1,2], array[3]];
+
+# array range access
+query T
+select array[1,NULL,2][-1:134124523];
+----
+{1,NULL,2}
+
+query T
+select (array[1,NULL,2])[3:1];
+----
+NULL
+
+query T
+select array[array[1],array[2],array[3]][-21432315:134124523];
+----
+{{1},{2},{3}}
+
+query T
+select array[array[1],array[2],array[3]][-21432315:134124523][1];
+----
+{1}
+
+query T
+select array[array[1],array[2],array[3]][-21432315:134124523][1:2];
+----
+{{1},{2}}
+
+query T
+select array[array[1],array[2],array[3]][-21432315:134124523][1];
+----
+{1}
+
+query I
+select array[array[1],array[2],array[3]][-21432315:134124523][1][1];
+----
+1
+
+query T
+select (array[1,NULL,2])[:3];
+----
+{1,NULL,2}
+
+query T
+select (array[1,NULL,2])[:2];
+----
+{1,NULL}
+
+query T
+select (array[1,NULL,2])[:1];
+----
+{1}
+
+query T
+select (array[1,NULL,2])[:999];
+----
+{1,NULL,2}
+
+# TODO: PG returns `{}` in this case
```

Comment:
just don't understand the rational behind this behavior
Isn't it better to make `[:0]` and `[:-1]` return the same thing, aka NULL?
they both go out of range

Reply:
https://www.postgresql.org/docs/current/arrays.html
> An array slice expression likewise yields null if the array itself or any of the subscript expressions are null. However, in other cases such as selecting an array slice that is completely outside the current array bounds, a slice expression yields an empty (zero-dimensional) array instead of null. (This does not match non-slice behavior and is done for historical reasons.) If the requested slice partially overlaps the array bounds, then it is silently reduced to just the overlapping region instead of returning null.

PostgreSQL returns an empty array for both `[:0]` and `[:-1]`, and null for both `[0]` and `[-1]`. It also returns empty array `{}` for `[3:2]`.

Reply:
Oh, I see, I probably confused with some other test cases

Reply:
Now all these cases return `{}`.